crypto/x509/v3_asid.c

```diff
@@ -330,6 +330,7 @@ static int ASIdentifierChoice_is_canonical(ASIdentifierChoice *choice)
     ret = 1;
 
  done:
+    ASN1_INTEGER_free(old_max);
     ASN1_INTEGER_free(a_max_plus_one);
```

Comment:
Won't this double free old_max?

Reply:
Yes.  I've adopted your suggestion with `new_max` and delaying the assignment.